lib/gs-category.c

```diff
@@ -525,6 +525,27 @@ gs_category_sort_children (GsCategory *category)
 			  gs_category_sort_children_cb);
 }
 
+/**
+ * gs_category_equal:
+ * @a: a #GsCategory
+ * @b: a #GsCategory
+ *
+ * Returns: %TRUE if @a and @b have the same category id, and %FALSE otherwise.
+ **/
+gboolean
+gs_category_equal (GsCategory *a,
+		   GsCategory *b)
+{
+	const char *a_id, *b_id;
+
+	g_return_val_if_fail (GS_IS_CATEGORY (a), FALSE);
+	g_return_val_if_fail (GS_IS_CATEGORY (b), FALSE);
+
```

Comment:
I think it would be worth adding the pointer comparison here as well as a small optimization for when the category object did not actually change.

Reply:
Pushed a fixup